puzzles/forms.py

```diff
@@ -24,7 +24,7 @@ class Meta:
         fields = ["status"]
 
     status = forms.ChoiceField(
-        choices=[(status, status) for status in Puzzle.STATUS_CHOICES],
+        choices=[(status, status) for status in Puzzle.ALL_STATUSES],
```

Comment:
should this just be VISIBLE_STATUS_CHOICES?

Reply:
Also curious: how come "status" gets repeated twice: (status, status)? What is that for?

Reply:
@erwa -- https://docs.djangoproject.com/en/2.2/ref/models/fields/#choices

The first is the actual model field that gets set, while the second is the human-readable string shown in the UI. In our case, the two are the same.

@rawxfish -- Actually no. It's a bit counter-intuitive, but these is the essentially the default choices. In puzzle_extras.py, we restrict choices to the VISIBLE_STATUS_CHOICES if the right conditions are met. There are other ways to do this, but I felt this was the easiest.

Reply:
Thanks for the explanation.

Ah, so here we are defining the "ModelForm" so we need to list all possible values for status, whereas puzzle_extras.py is for rendering the template that gets shown to the user, so there we want to filter the choices.